draft/webxdc-dev-reference.md

```diff
@@ -21,13 +21,14 @@ no need to add `webxdc.js` to your ZIP-file):
 ### sendUpdate()
 
 ```js
-window.webxdc.sendUpdate(payload, descr);
+window.webxdc.sendUpdate(update, descr);
 ```
 
 Webxdc apps are usually shared in a chat and run independently on each peer.
 To get a shared state, the peers use `sendUpdate()` to send updates to each other.
 
-- `payload`: any javascript primitive, array or object.
+- `update`: an object with the following fields:  
+  `update.payload`: any javascript primitive, array or object.
```

Comment:
why use a nested object here, I think the previous idea of a function with 3 arguments, some of them optional is better JS ux when users don't use typescript - or do you want to allow to send updates that only change the summary without having a payload?

Reply:
it would not only be 3 arguments, already now we would have 4. and there are more already appearing (maybe an "is aggregate" flag, maybe a "fast" for p2p, whatever). so named parameters in an object seems to be easier to handle.

the idea is to allow sth. 
`sendUpdate({payload: ...})` as well as  
`sendUpdate({payload: ..., summary: ... })`,  
`sendUpdate({payload: ..., summary: ... msg: ... })`  
`sendUpdate({payload: ..., msg: ... })`  
maybe even payload may be left out.

there are other options ([some thoughts](https://gist.github.com/r10s/f5a2aec6c8a32b0749bbdd1ce721738e)), however, that one seems to be the best one.

also, a similar structure is already used on receiving, so it seems to be nice code to just use the same here.

Reply:
> wrt ux: i think, esp. when using nothing specialized and you do not have the help of an ide, this is easier to read.

ok, maybe we should add a runtime check somehow so that the method throws if the parameters are not set correctly. or later just say that you can use the typescript definition file to check if your function calls are correct.

Reply:
if "msg" is added to show a system message to dc users, that msg could be used for the description for other MUAs as well if the desc parameter is omitted

Reply:
> if "msg" is added to show a system message to dc users, that msg could be used for the description for other MUAs as well if the desc parameter is omitted, but I like it as is already, really nice!

nice idea! we can implement it like that in the subsequent pr.